collect_app/src/main/java/org/odk/collect/android/utilities/TextUtils.java

```diff
@@ -97,7 +103,48 @@ public static CharSequence textToHtml(String text) {
         }
 
         return Html.fromHtml(markdownToHtml(text));
+    }
+
+    public static String compress(String data) throws IOException {
+        if (data == null || data.length() == 0) {
+            return data;
+        }
 
+        // Encode string into bytes
+        byte[] input = data.getBytes("UTF-8");
+
+        // Compress the bytes
+        byte[] output = new byte[input.length];
+        Deflater compresser = new Deflater();
+        compresser.setInput(input);
+        compresser.finish();
+        int compressedDataLength = compresser.deflate(output);
+        compresser.end();
+
+        // Encode to base64
+        String base64String = Base64.encodeBase64String(output);
+        Timber.i("Original length : %d", data.length());
+        Timber.i("Compressed length : %d", compressedDataLength);
+        Timber.i("Compression ratio : %2f", ((data.length() * 1.0) / compressedDataLength) * 100);
+        return base64String;
     }
 
+    public static String decompress(String compressedString) throws IOException, DataFormatException {
+
+        // Decode from base64
+        byte[] output = Base64.decodeBase64(compressedString);
+
+        // Decompresses the bytes
+        Inflater decompresser = new Inflater();
+        decompresser.setInput(output);
+        byte[] result = compressedString.getBytes();
+        int resultLength = decompresser.inflate(result);
+        decompresser.end();
+
+        // Decode the bytes into a String
+        String outputString = new String(result, 0, resultLength, "UTF-8");
+        Timber.i("Compressed length : %d", compressedString.length());
+        Timber.i("Decompressed length : %d", resultLength);
+        return outputString;
+    }
```

Comment:
I also don't see any test coverage of these methods. Could you add a couple simple unit tests?